### PR TITLE
[WIP] AEROGEAR-1924 CLI service integration

### PR DESCRIPTION
**Pull request by dimitraz**
**Describe what this PR does and why we need it**:
Review existing command for integrating mobile services

Changes proposed in this pull request
 - Add tests for integration commands
 - Add a watch on the binding resource and report on its status 
 - Add a `--no-wait` to exit immediately 
- Review documentation

**To do**
- Need to properly test the integration and watch command, currently waiting on [this](https://github.com/aerogearcatalog/keycloak-apb/issues/28#issuecomment-360797428) issue

**Which issue this PR fixes (This will close that issue when PR gets merged)**
[AEROGEAR-1924](https://issues.jboss.org/browse/AEROGEAR-1924)

### CODE REVIEW

File: pkg/cmd/integration.go [hunk 1]
```diff
@@ -122,51 +123,88 @@ oc plugin mobile create integration <consuming_service_instance_id> <providing_s
...
 				return errors.Wrap(err, "failed to get deployment for service "+consumerSvcInstName)
 			}
 			dep.Spec.Template.Labels[providerSvc.Name] = "enabled"
 			if _, err := bc.k8Client.AppsV1beta1().Deployments(namespace).Update(dep); err != nil {
-				return errors.Wrap(err, "failed up update deployment for "+consumerSvcInstName)
+				return errors.Wrap(err, "failed to update deployment for service "+consumerSvcInstName)
+			}
+
+			// exit immediately
+			noWait, err := cmd.PersistentFlags().GetBool("no-wait")
```
Comment (maleck13) [inline pkg/cmd/integration.go]: I think the logic here should go:

```
if noWait && ! redeploy{
  return 
}
//wait logic

if redeploy{
//redeploy logic
}

```

If you want me to assist here  I am happy to do so

File: pkg/cmd/integration.go [hunk 2]
```diff
@@ -107,12 +108,12 @@ func (bc *IntegrationCmd) CreateIntegrationCmd() *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "integration <consuming_service_instance_id> <providing_service_instance_id>",
 		Short: "integrate certain mobile services together",
-		Long: `example usage: kubectl plugin mobile create integration <consuming_service_instance_id> <providing_service_instance_id>
```
Comment (maleck13) [inline pkg/cmd/integration.go]: I think in the long description we should give an example of how to get the serviceinstance ids

``` 
mobile get serviceinstances fh-sync-server
mobile get serviceinstances keycloak
```

File: pkg/cmd/integration_test.go [hunk 3]
```diff
@@ -0,0 +1,516 @@
...
+				t.Fatal("expected an error but got none")
+			}
+			if tc.ValidateErr != nil {
+				tc.ValidateErr(t, err)
+			}
+		})
+	}
+}
+
+func TestIntegrationCmd_GetIntegrationCmd(t *testing.T) {
```
Comment (maleck13) [inline pkg/cmd/integration_test.go]: I think we can remove this unless you are intending to fill it in

**Comment by maleck13** [comment]
@dimitraz can you rebase pls

File: pkg/cmd/integration.go (hunk 1 above)
Comment (dimitraz) [reply pkg/cmd/integration.go]: What happens if a user specifies `no-wait` and `redeploy`? I'm guessing that's not possible so I could just send back the please redeploy message?

Comment (maleck13) [reply pkg/cmd/integration.go]: As mentioned on IRC we have to wait if auto redeploy is set. We will document this in the command docs

**Comment by dimitraz** [comment]
Closing, see https://github.com/aerogear/mobile-cli/pull/58

**Pull request closed without merging**